code/game/mecha/mecha.dm

```diff
@@ -674,7 +678,92 @@
 		if(src_object in view(2, src))
 			return STATUS_UPDATE //if they're close enough, allow the occupant to see the screen through the viewport or whatever.
 
-/obj/mecha/proc/melee_action(atom/target)
+/obj/mecha/proc/melee_action(atom/T)
+	if(internal_damage&MECHA_INT_CONTROL_LOST)
+		T = safepick(oview(1,src))
+	if(!melee_can_hit)
+		return
+
+	if(istype(T, /mob/living))
+		var/mob/living/M = T
+		if(src.occupant.a_intent == I_HURT || istype(src.occupant, /mob/living/carbon/brain)) //Brains cannot change intents; Exo-piloting brains lack any form of physical feedback for control, limiting the ability to 'play nice'.
+			playsound(src, 'sound/weapons/heavysmash.ogg', 50, 1)
+			do_attack_animation(T)
+			if(damtype == "brute")
+				step_away(M,src,15)
+
+			if(ishuman(T))
+				var/mob/living/carbon/human/H = T
+
+				var/obj/item/organ/external/temp = H.get_organ(pick(BP_TORSO, BP_TORSO, BP_TORSO, BP_HEAD))
+				if(temp)
+					var/update = 0
+					switch(damtype)
+						if("brute")
+							H.Paralyse(1)
+							update |= temp.take_damage(rand(force/2, force), 0)
+						if("fire")
+							update |= temp.take_damage(0, rand(force/2, force))
+						if("tox")
+							if(H.reagents)
+								if(H.reagents.get_reagent_amount("carpotoxin") + force < force*2)
+									H.reagents.add_reagent("carpotoxin", force)
+								if(H.reagents.get_reagent_amount("cryptobiolin") + force < force*2)
+									H.reagents.add_reagent("cryptobiolin", force)
+						if("halloss")
+							H.stun_effect_act(1, force / 2, BP_TORSO, src)
+						else
+							return
+					if(update)	H.UpdateDamageIcon()
+				H.updatehealth()
+
+			else
+				switch(damtype)

```

Comment:
This pains me to read but the only way I can conceive of fixing it is barely any more readable